db/database.go

```diff
@@ -68,7 +68,10 @@ var (
 )
 
 var DefaultCompactInterval = uint32(60 * 60 * 24) // Default compact interval in seconds = 1 Day
-var DefaultResyncQueryLimit = 5000
+var (
+	DefaultResyncQueryLimit    = 5000
+	DefaultPrincipalQueryLimit = 5000
```

Comment:
Having three different query pagination config options, all with the same default, feels like too granular a level customization.  I don't know if users really need the ability to set the channel query pagination limit to a value that's different than the one they use for resync or principal queries.  Given that, this feels like it's just adding bloat to the configuration.

Reply:
Yeah thats a fair comment. Its probably fine to narrow this down to just be one 'QueryPaginationLimit'. Like you say, not sure there is any reason why you'd want pagination limit to be different for each one.